spring-cloud-starter-stream-common-tensorflow/src/main/java/org/springframework/cloud/stream/app/tensorflow/processor/TensorflowCommonProcessorConfiguration.java

```diff
@@ -146,29 +135,51 @@ public TensorFlowService tensorFlowService() throws IOException {
 	@ConditionalOnMissingBean(name = "tensorflowOutputConverter")
 	public TensorflowOutputConverter tensorflowOutputConverter() {
 		// Default implementations serializes the Tensor into Tuple
-		return new TensorflowOutputConverter<Tuple>() {
-
-			@Override
-			public Tuple convert(Tensor tensor, Map<String, Object> processorContext) {
-				return TensorTupleConverter.toTuple(tensor);
-			}
+		return (TensorflowOutputConverter<Tuple>) (tensorMap, processorContext) -> {
+			Tensor<?> tensor = tensorMap.entrySet().iterator().next().getValue();
+			return TensorTupleConverter.toTuple(tensor);
 		};
 	}
 
 	@Bean
 	@ConditionalOnMissingBean(name = "tensorflowInputConverter")
 	@SuppressWarnings("unchecked")
 	public TensorflowInputConverter tensorflowInputConverter() {
-		return new TensorflowInputConverter() {
+		return (input, processorContext) -> {
+
+			if (input instanceof Map) {
+				return (Map<String, Object>) input;
+			}
+			else if (input instanceof Tuple) {
+				Tuple tuple = (Tuple) input;
+				Map<String, Object> map = new LinkedHashMap<>(tuple.size());
+				for (int i = 0; i < tuple.size(); i++) {
+					map.put(tuple.getFieldNames().get(i), tuple.getValue(i));
+				}
+				return map;
+			}
+
+			throw new MessageConversionException("Unsupported input format: " + input);
+		};
+	}
 
+	@Bean
+	@StreamMessageConverter
+	public MessageConverter toTensorMessageConverter() {
+		return new AbstractMessageConverter(new MimeType("application", "x-tensor")) {
```

Comment:
Where this converter is used?

I thought we need them when we convert `input`/`output` from/to external world, therefore it is a bit confusing for me why would one need this custom mime-type?

Thanks

Reply:
Good spot. I believe this is a left over from my experiments with the new content-type support. I will remove it